src/Compilers/CSharp/Portable/Compilation/DeconstructionInfo.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    // Deconstructions are represented internally as a tree of conversions, but
+    // since they are not conversions from the language perspective we use a wrapper to
+    // abstract the public API from the implementation.
+    public struct DeconstructionInfo : IEquatable<DeconstructionInfo>
+    {
+        private readonly Conversion _conversion;
+
+        public IMethodSymbol Method
+        {
+            get
+            {
+                return _conversion.Kind == ConversionKind.Deconstruction
```

Comment:
Why would we ever have a DeconstructionInfo where the _conversion wasn't a Deconstruction to start with? Is this for default-initialized structs? If so, add a comment please?

Reply:
`(long x, long y) = e;` with `E.Deconstruct(out int, out int)` is represented as a conversion (kind=`Deconstruction`) with a reference to this `Deconstruct` method, and two nested conversions (kind=`ImplicitNumeric`) from `int` to `long`.
Such trees can become deeper for nested deconstructions.

Reply:
Can you add documentation to explain this? We should give our API callers some hope.